tracer.go

```diff
@@ -121,6 +126,110 @@ type StartSpanOptions struct {
 	// identical to those for Span.SetTag(). May be nil.
 	//
 	// If specified, the caller hands off ownership of Tags at
-	// StartSpanWithOptions() invocation time.
+	// StartSpan() invocation time.
 	Tags map[string]interface{}
 }
+
+// StartSpanOption instances (zero or more) may be passed to Tracer.StartSpan.
+type StartSpanOption interface {
+	Apply(*StartSpanOptions)
+}
+
+// ReferenceType is an enum type describing different categories of
+// relationships between two Spans. If Span-2 refers to Span-1, the
+// ReferenceType describes Span-1 from Span-2's perspective. For example,
+// RefBlockedParent means that Span-1 caused Span-2, and that it's blocked on
+// Span-2's finish.
+type ReferenceType int
+
+const (
+	// RefBlockedParent refers to a "parent" Span that (a) caused the creation
+	// of the newly-started span, and (b) is blocked on the newly-started Span.
+	// See RefStartedBefore for the non-blocking version. Timing diagram:
+	//
+	//     [-Referent------------]
+	//               [-New Span-]
+	//
+	RefBlockedParent ReferenceType = iota
+
+	// RefRPCClient is a special case of RefBlockedParent for the server side
+	// of RPCs to refer to the client side of RPCs. Timing diagram:
+	//
+	//     [-Remote Client Referent------]
+	//                [-New Server Span-]
+	//
+	RefRPCClient
+
+	// RefStartedBefore refers to a Span that merely started before the
+	// newly-started Span. This is the weakest of the ReferenceType causality
+	// assertions. Timing diagram:
+	//
+	//     [-Referent-]
+	//          [-New Span-]
+	//
+	RefStartedBefore
+
+	// RefFinishedBefore refers to a Span that finished (and started) before
+	// the newly-created Span. Timing diagram:
+	//
+	//     [-Referent-]
+	//                    [-New Span-]
+	//
+	RefFinishedBefore
```

Comment:
similarly, maybe mention as example that this ref type can be used to model joins.